tests/specification/mod.rs

```diff
@@ -0,0 +1,70 @@
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+mod reports;
+mod run;
+mod test_errors;
+
+#[test_log::test]
+pub fn spec_dummy() {
+    let path = "./tests/specification/dummy.wast";
+    let report = run::run_spec_test(path);
+    println!("Report for {}:\n{}", path, report);
+}
+
+#[test_log::test]
+pub fn spec_tests() {
+    let paths = get_wast_files(Path::new("./tests/specification/testsuite/"))
+        .expect("Failed to find testsuite");
+
+    let mut successful_reports = 0;
+    let mut failed_reports = 0;
+    let mut compile_error_reports = 0;
+
+    for test_path in paths {
+        let report = run::run_spec_test(test_path.to_str().unwrap());
+        println!("Report for {}:\n{}", test_path.display(), report);
+
+        match report {
+            reports::WastTestReport::Asserts(assert_report) => {
+                if assert_report.has_errors() {
+                    failed_reports += 1;
+                } else {
+                    successful_reports += 1;
+                }
+            }
+            reports::WastTestReport::CompilationError(_) => {
+                compile_error_reports += 1;
+            }
+        }
+    }
+
+    println!(
```

Comment:
would it be  clever to use the `log` crates macro here, since we already activated them? That way, one can make the `cargo test` call noisy on demand, just by saying `RUST_LOG=info cargo test`?

Reply:
+1 for `log`. We could use `info!(...)` or `warn!(...)` depending on whether there were errors?

Reply:
I'm going to push against this, not because it wouldn't be nice, but because it's too much of a hassle. The env_log crate puts it's own header before every message, I would need to refactor the print method from a `impl Display` to a standalone function to pick and choose what gets displayed. This isn't unsolvable, but I want to do that in another PR.